src/Horus/FunctionAnalysis.hs

```diff
@@ -108,8 +111,8 @@ cgMbInsertArc (CG verts arcs) (fro, to) =
     then Nothing
     else Just . CG verts $ Map.insertWith (++) fro [to] arcs
 
-graphOfCG :: CG -> (Graph, Vertex -> (Label, Label, [Label]))
-graphOfCG cg = graphFromEdges' . map named . Map.assocs $ cg_arcs cg
+graphOfCG :: CG -> (Graph, Vertex -> (Label, Label, [Label]), Label -> Maybe Vertex)
+graphOfCG cg = graphFromEdges . map named . Map.assocs $ cg_arcs cg
```

Comment:
The return type of this function is quite... complicated. For example, there is this tuple inside of type `(Label, Label, [Label])`. What is that?

How about a docstring describing what you get back and perhaps what it would be useful for, and possibly a more verbose name that captures what exactly the function is giving you.

Where it is used, the names sort of suggest that the last two tuple elements are maps or functions (or at least have the same kind) but this is not the case.

Reply:
I am torn on this. I understand that it's complicated but this is coming from Data.Graph essentially 🤔 .

Reply:
I am only suggesting adding a comment here. You feel that it is transparent what is going on?

Reply:
Nono, I feel that commenting what is essentially a forward on a std function is strange; isn't it?

Reply:
It is, I did not read this carefully enough and assumed that the return type from `graphFromEdges` was nontrivially affected by what we were passing it as arguments. It is not, we just have that `node` and `key` are both `Label` in our case. I was being dumb!